src/custom_fourier_types.jl

```diff
@@ -1,20 +1,31 @@
-##  This View checks for the index to be L1 or the mirrored version (L2)
-# and then replaces the value by half of the parent at L1
+"""
+FourierSplit{T,N, AA<:AbstractArray{T, N}} <: AbstractArray{T,N}
+
+This View checks for the index to be L1 or the mirrored version (L2)
+and then replaces the value by half of the parent at L1
+`do_split` is a Bool that indicates whether this mechanism is active. 
+It is needed for type stability reasons of functions returnting this type.
+"""
 struct FourierSplit{T,N, AA<:AbstractArray{T, N}} <: AbstractArray{T,N}
     parent::AA # holds the data (or is another view)
     D::Int # dimension along which to apply to copy
     L1::Int # low index position to copy from (and half)
     L2::Int # high index positon to copy to (and half)
+    do_split::Bool
+
+"""
+    FourierSplit(parent::AA, D::Int,L1::Int,L2::Int, do_split::Bool) where {T,N, AA<:AbstractArray{T, N}}
 
-    # This version below is needed to avoid a split for the firs rft dimension but still return half the value
-    # FFTs and other RFT dimension should use the version without L2
-    function FourierSplit(parent::AA, D::Int,L1::Int,L2::Int) where {T,N, AA<:AbstractArray{T, N}}
-        return new{T,N, AA}(parent, D, L1, L2)
+This version below is needed to avoid a split for the first rft dimension,
```

Comment:
Tab the doc string in the case, I guess.

Reply:
should be fine now